File: lab/example/csharp/src/DocSearch/Embedding/Embedder.cs
// Embedder-Port (ADR-0002). Importiert nur Types.
using System.Security.Cryptography;
using System.Text;
using DocSearch.Types;

namespace DocSearch.Embedding;

public interface IEmbedder
{
    float[] Embed(string text);
}

public sealed class MockEmbedder : IEmbedder
{
    public float[] Embed(string text)
    {
        byte[] digest = SHA256.HashData(Encoding.UTF8.GetBytes(text));
        long seed = 0L;
        for (int i = 0; i < 8; i++)
        {
            seed = (seed << 8) | digest[i];
        }
        float[] result = new float[Constants.EmbeddingDim];
        for (int i = 0; i < Constants.EmbeddingDim; i++)
        {
            seed = seed * 1103515245L + 12345L;
            result[i] = (float)(((seed >> 16) & 0x7FFFL) / 32768.0);
        }
        return result;
    }
}

File: lab/example/python/src/docsearch/embedding/embedder.py
"""Embedder-Adapter — Port aus ADR-0002."""

from __future__ import annotations

import hashlib
from typing import Protocol

from docsearch.types import EMBEDDING_DIM


class Embedder(Protocol):
    def embed(self, text: str) -> tuple[float, ...]:
        ...


class MockEmbedder:
    """Deterministisches Pseudo-Embedding: gleicher Text → gleicher Vektor (LH-QA-02)."""

    def embed(self, text: str) -> tuple[float, ...]:
        digest = hashlib.sha256(text.encode("utf-8")).digest()
        # Deterministische LCG-Sequenz auf Basis des Hashes.
        seed = int.from_bytes(digest[:8], "big")
        out: list[float] = []
        for _ in range(EMBEDDING_DIM):
            seed = (seed * 1103515245 + 12345) & 0xFFFFFFFFFFFFFFFF
            out.append(((seed >> 16) & 0x7FFF) / 32768.0)
        return tuple(out)
